consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py

```diff
@@ -107,4 +121,75 @@ def compare_forks(self, cur_fork_head, new_fork_head):
                 cur_fork_head.header_signature[-8:],
                 new_fork_head.header_signature[:8],
                 new_fork_head.header_signature[-8:])
-            return False
+            chosen_fork_head = cur_fork_head
+
+        # Now that we have chosen a fork for the chain head, if
+        # we chose the new fork, we need to create consensus state
+        # store information for the new fork's chain head.
+        if chosen_fork_head == new_fork_head:
+            # Get the state view for the previous block in the chain so we can
+            # create a PoET enclave
+            previous_block = None
+            try:
+                previous_block = \
+                    self._block_cache[new_fork_head.previous_block_id]
+            except KeyError:
+                pass
+
+            state_view = \
+                BlockWrapper.state_view_for_block(
+                    block_wrapper=previous_block,
+                    state_view_factory=self._state_view_factory)
+
+            poet_enclave_module = \
+                factory.PoetEnclaveFactory.get_poet_enclave_module(state_view)
```

Comment:
Is there an issue if the previous block isn't a PoET block?
I think later on it may raise an attribute error when we attempt to deserialize the wait certificate if this module is None.